packages/typography/src/Link/Link.tsx

```diff
@@ -40,21 +53,22 @@ const Link = Polymorphic<LinkProps>(
     }, []);
 
     const { theme } = useDarkMode(darkModeProp);
-    const { Component } = usePolymorphic(as ? as : href ? 'a' : 'span');
+    const baseFontSize = useUpdatedBaseFontSize(baseFontSizeOverride);
+    const { Component } = useInferredPolymorphic(as, rest, 'span');
 
     const hrefHostname = useMemo(() => {
-      if (!href) return;
-      const httpRegex = /^http(s)?:\/\//;
-      return httpRegex.test(href) ? new URL(href).hostname : currentHostname;
-    }, [href, currentHostname]);
-
-    const baseFontSize = useUpdatedBaseFontSize(baseFontSizeOverride);
+      if (hasAnchorLikeProps(rest)) {
+        const httpRegex = /^http(s)?:\/\//;
+        return httpRegex.test(rest.href)
+          ? new URL(rest.href).hostname
+          : currentHostname;
+      }
+    }, [rest, currentHostname]);
 
-    let target, icon, rel;
+    let target, rel, icon;
 
-    if (targetProp) {
-      target = targetProp;
-    } else {
+    // Sets defaults for target and rel props when Component is an anchor tag
+    if (Component === 'a') {
```

Comment:
Nit, but I don't know if I like these uninitialized variables here. I'd prefer see something like below, or even a function that returns these default props. 
```js
const defaultAnchorProps = {
  target: <logic>,
  rel: <logic>
}
```